lms/djangoapps/instructor_task/docs/decisions/001-scheduling-instructor-tasks.rst

```diff
@@ -0,0 +1,64 @@
+==========================
+Scheduled Instructor Tasks
+==========================
+
+Status
+------
+
+Draft
+
+Background
+----------
+
+One of the most requested features in edx-platform has been the ability to schedule an email message to be sent at a specified date and time through the Instructor Dashboard's Bulk Course Email tool. These email messages are sent as an instructor task. Today, edx-platform does not have the ability to support scheduling of instructor tasks. When an instructor task is created we attempt to execute the task immediately.
+
+This ADR covers details on the proposed modeling of a scheduled instructor task. A subsequent ADR will focus on the processing of scheduled instructor tasks.
+
+Decision
+--------
+
+* We will introduce a new ``InstructorTaskSchedule`` model to track the (optional) schedule of an instructor task.
+* We will extend the existing ``InstructorTask`` model with a new (nullable) column to link an instructor task to its schedule.
+* We will introduce a new **SCHEDULED** state that will be used by the ``InstructorTask`` model to denote a task that has been scheduled for execution at a later date and time.
+* Instructor tasks without a schedule will continue to be executed immediately.
+* A scheduled instructor task will be reserved in the LMS database but will *not* be submitted to Celery for execution until it is due.
+
+The ``InstructorTaskSchedule`` model
+====================================
+
+This new model that will be responsible for tracking the due date of an instructor task, as well as some information needed to process the tasks later. The model will have two fields:
+
+* **due** (DateTime): The date and time (in UTC) to execute the associated task.
+* **task_args** (TextField): This will store information required to execute the task when due. The source data is in the form of a dictionary that will be converted to text for storage in the database.
+
+Extending the ``InstructorTask`` model
+======================================
+
+We will extend the existing ``InstructorTask`` model with a new nullable column named **schedule**. This field will have a OneToOne relation to the ``InstructorTaskSchedule`` model. An instructor task is a one-time task and thus should only ever have a single schedule attributed to it.
```

Comment:
Have you considered putting the association on the `InstructorTaskSchedule` model instead of this one?

[From these Django docs](https://docs.djangoproject.com/en/4.0/topics/db/examples/one_to_one/) I got the impression that the "optional" model is the one to hold the reference to its "parent". And in this case, I’m thinking of the `InstructorTask` as the "parent" with the `InstructorTaskSchedule` as its optional "child".

I have zero idea if that matters at all, but that would prevent needing the `nullable` column on `InstructorTask`.

Reply:
@oliviaruizknott I missed this comment until today! Thanks for the doc link and the good question. I may still have some crossed wires here in my thinking/implementation.